### PR TITLE
data/validation: add time format validator

### PR DESCRIPTION
**Pull request by Jacalz**
### Description:

Adds a validator that I used in [Sparta](https://github.com/Jacalz/sparta) for verifying that dates and times were correct.
Can probably be useful for a lot of other use cases as well :)

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.
- [x] Public APIs match existing style.

### CODE REVIEW

File: data/validation/time.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// The validator will return nil if valid, otherwise returns an error with a reason text.
+//
+// Since: 2.1
+func NewTime(format string) fyne.StringValidator {
+	return func(text string) error {
+		if _, err := time.Parse(format, text); err != nil {
+			return err
+		}
+
+		return nil
```
Comment (stuartmscott) [inline data/validation/time.go:20]:
```suggestion
		_, err := time.Parse(format, text)
		return err
```

**Review by stuartmscott — CHANGES_REQUESTED** [review]
A useful addition Jacob!

File: data/validation/time.go (hunk 1 above)
Comment (Jacalz) [reply data/validation/time.go]: Good idea. Thanks 🙂

**Review by stuartmscott — APPROVED** [review]

**Review by andydotxyz — COMMENTED** [review]
Nice new Validator. 
mid suggest linking to the date format docs though, the Go team made a really good job of describing various formats.

**Review by stuartmscott — DISMISSED** [review]

**Review by andydotxyz — APPROVED** [review]

**Comment by fpabl0** [comment]
Although it is good to have the link to the go docs. I would prefer to keep this line too:
```
The reference time for the format: Mon Jan 2 15:04:05 -0700 MST 2006.
```
Sometimes, you already know how it works, but you do not remember the reference time, and opening a link just to see it does not seem to be a good experience, does it?

**Comment by andydotxyz** [comment]
> Although it is good to have the link to the go docs. I would prefer to keep this line too:

That is a good point, I agree

**Review by andydotxyz — CHANGES_REQUESTED** [review]
I like @fpabl0 comment about re-adding the reference line for quicker doc lookup for the familiar.

**Comment by Jacalz** [comment]
Good point. Added it back :slightly_smiling_face:

**Review by andydotxyz — APPROVED** [review]
Great thanks

File: data/validation/time_test.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import (
+	"testing"
+
+	"fyne.io/fyne/v2/data/validation"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTime(t *testing.T) {
+	time := validation.NewTime("2006-01-02")
```
[the remark this answers is not in the document]
Comment (Jacalz) [reply data/validation/time_test.go]: Good point. I renamed it to `validate`.

**Review by fpabl0 — APPROVED** [review]
LGTM :)

**Pull request merged**